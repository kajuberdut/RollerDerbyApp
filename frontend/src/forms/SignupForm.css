.SignupForm {

    margin-top: 10%;
    margin-left: 20%;
    width: 40%;
    
}

