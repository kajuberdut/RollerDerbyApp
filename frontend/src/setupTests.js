// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';


global.localStorage = {
  getItem: jest.fn(),
  setItem: jest.fn(),
};

const user = {
  userId: 1,
  username: "testUser",
  email: "testuser@gmail.com",
  facebookName: "facebook test user",
  about: "This is my story...",
  primaryNumber: "01",
  level: "A",
  ruleset: [],
  position: [],
  insurance: [],
  locationId: null,
  associatedLeagues: "Testing League",
  phoneNumber: "1111111111",
  firstName: "Jane",
  lastName: "Doe",
  additionalInfo: "Epi pen in bag.",
  secondaryNumber: "02"
};

beforeEach(() => {
  localStorage.setItem("user", JSON.stringify(user));
});

