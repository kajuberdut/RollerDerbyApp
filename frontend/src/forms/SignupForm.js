import React, { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import "./SignupForm.css"
import { Card, CardBody, CardTitle, Form, FormGroup, Label, Input, Button } from "reactstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEye } from "@fortawesome/free-solid-svg-icons";
import { faEyeSlash } from "@fortawesome/free-solid-svg-icons";

/** 
 * Display signup form 
 */

const SignupForm = ({signup, setIsSignupVis}) => {

  /** Set error message in state */
  const [errorMessage, setErrorMessage] = useState([]);
  const navigate = useNavigate();
  
  /** Sets initial state of form   */

  let INITIAL_STATE = { username: "",  email: "", password: ""};

  /** Sets formData in initial state */

  const [formData, setFormData] = useState(INITIAL_STATE);
  const [passVis, setPassVis] = useState(false);


  /** When login page is mounted setIsloginVis to true - false when unmounted */

  useEffect(() => {
    setIsSignupVis(true);
  return () => {
    setIsSignupVis(false);
  };
  }, []);


  /** Handle Submit by either creating user or returning an error message */

  const handleSubmit = async evt => {
    evt.preventDefault();   
    setFormData(INITIAL_STATE);

    let result = await signup(formData);
    console.log("result", result)

    if(result.success) {
        navigate('/login')
    } else {
      setErrorMessage(result.err)
    }
  }

  /** Update local state with current state of input element */

  const handleChange = evt => {
    console.log('handleChange is running')
    const { name, value }= evt.target;

    setFormData(fData => ({
      ...fData,
      [name]: value,
    }));

  };

  /** Toggle password visibility */

  const togglePassVis = () => {
    setPassVis(!passVis);
  };


  /** render signup form form */

  return (
    <section className="col-md-4 SignupForm" style={{minWidth: '400px'}}>
        <Card>
            <CardTitle className="SignupForm-CardTitle">
            <h1>Create a profile</h1>
            <div>
            {errorMessage && (<div style={{color: 'red'}}>{errorMessage}</div>)}
            </div>
            </CardTitle>
            <CardBody>
                <Form onSubmit={handleSubmit}>
                    <FormGroup>
                        <Label htmlFor="username" sm={10}>Derby Name: </Label>
                
                        <Input
                            id="username"
                            name="username"
                            defaultValue={formData.username}
                            onChange={handleChange}
                            placeholder="Derby Name"
                            required
                            maxLength={35}
                        />
                  
                        <Label htmlFor="email">Email: </Label>
                        <Input
                            type="email"
                            id="email"
                            name="email"
                            value={formData.email}
                            onChange={handleChange}
                            placeholder="Email"
                            autoComplete="username"
                            required
                            maxLength={128}
                        />

                        <Label htmlFor="password">Password: </Label>
                      <div style={{display: "flex"}}>
                        
                          <Input
                              type={passVis ? "text" : "password"}
                              name="password"
                              value={formData.password}
                              onChange={handleChange}
                              placeholder="Password"
                              id="password"
                              autoComplete="new-password"
                              required
                              maxLength={128}
                              minLength={8}
                              style={{ width: "85%" }}
                          />
                          <div className='EyeIcon' role="EyeIcon" onClick={togglePassVis} style={{paddingLeft: '10px'}}>
                            <FontAwesomeIcon icon={passVis ? faEye : faEyeSlash} className='EyeIcon-Icon'/>
                          </div>
                        </div>
                    </FormGroup>
                    <Button >Create Profile</Button>
                </Form>
            </CardBody>
        </Card>
    </section>
  );
};

export default SignupForm;
