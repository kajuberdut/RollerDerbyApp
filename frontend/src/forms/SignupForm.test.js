import { render, screen, cleanup } from '@testing-library/react';
import SignupForm from './SignupForm';
import { BrowserRouter } from "react-router-dom";
import '../setupTests.js';

// * Passing as of 2/13/24
// npm test SignupForm.test.js
// must be in frontend directory 

afterEach(() => {
    cleanup()
  })
  
describe("checks rendering of signup form", () => {  

    test('renders signup form page', () => {

    render(
        <BrowserRouter>
                <SignupForm setIsSignupVis={jest.fn()} />
        </BrowserRouter>);
    });


    test('matches snapshot', function() {
    const { asFragment } = render(
        <BrowserRouter>
                <SignupForm setIsSignupVis={jest.fn()} />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot(); 
    })

});

describe("asserts signup form text", () => {  

    test('renders signup form page', () => {

    render(
        <BrowserRouter>
                <SignupForm setIsSignupVis={jest.fn()} />
        </BrowserRouter>
    )

        const text = screen.getByText('Create a profile');
        expect(text).toBeInTheDocument();
    });

});