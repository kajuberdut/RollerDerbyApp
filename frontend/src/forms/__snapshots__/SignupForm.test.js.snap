// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks rendering of signup form matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4 SignupForm"
    style="min-width: 400px;"
  >
    <div
      class="card"
    >
      <div
        class="SignupForm-CardTitle card-title"
      >
        <h1>
          Create a profile
        </h1>
        <div>
          <div
            style="color: red;"
          />
        </div>
      </div>
      <div
        class="card-body"
      >
        <form
          class=""
        >
          <div
            class="mb-3"
          >
            <label
              class="col-sm-10 col-form-label"
              for="username"
            >
              Derby Name: 
            </label>
            <input
              class="form-control"
              id="username"
              maxlength="35"
              name="username"
              placeholder="Derby Name"
              required=""
              type="text"
              value=""
            />
            <label
              class="form-label"
              for="email"
            >
              Email: 
            </label>
            <input
              autocomplete="username"
              class="form-control"
              id="email"
              maxlength="128"
              name="email"
              placeholder="Email"
              required=""
              type="email"
              value=""
            />
            <label
              class="form-label"
              for="password"
            >
              Password: 
            </label>
            <input
              autocomplete="new-password"
              class="form-control"
              id="password"
              maxlength="128"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-secondary"
          >
            Create Profile
          </button>
        </form>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
